sdk/src/main/scala/com/horizen/SidechainState.scala

```diff
@@ -266,9 +267,19 @@ class SidechainState private[horizen] (stateStorage: SidechainStateStorage,
         }
       }
 
-      newCoinsBoxesAmount = tx.newBoxes().asScala
+      var newCoinsBoxesAmount: Long = 0L
```

Comment:
let's separate `newCoinsBoxesAmount` calculation and forgers checks.
Even if it takes additional loop through the box list, still better to avoid mixing these two quite different operations.

Reply:
Ok done!